docs/shared-settings.md

```diff
@@ -60,6 +60,26 @@ If a path is relative, it will be resolved from CWD.
 { "resolvePaths": [] }
 ```
 
+## resolverConfig
+
+Override the options generated by this plugin for resolving require and import statements. These
+options are passed down to `enhanced-resolve`'s factory method and are documented [here on their
+GitHub repository](https://github.com/webpack/enhanced-resolve?tab=readme-ov-file#resolver-options).
```

Comment:
I'm a bit concerned that this will over-bind this package to `enhanced-resolve`, which may make it harder for us to make changes in the future - e.g. moving to `import.meta.resolve` (which I remember it has been discussed somewhere before).

We can state the minimum necessary in the docs. wdty?

Reply:
I guess the minimum necessary really depends on everyone's specific use cases. For us (and why I made this PR) it would be specifying  `resolverConfig.modules` to add an additional path for modules lookup. For another user it might be any of the 25 options documented in `enhanced-resolve`'s README.

Or did you mean something else by the minimum neccesary?

Reply:
yes, I think `resolverConfig.modules` is enough for now. I don't feel that the implementation has to be changed as well - other options can be something like undocumented features. thoughts? /cc @scagood

Reply:
@scagood your insights are wanted here. :)

Reply:
:thinking: I am on the fence here, because `enhanced-resolve` covers typescripts silly resolution options far better than `import.meta.resolve` -- I feel like we may need to re-write large chunks of the resolver to make it work.

That being said, I am happy to leave this as documenting only `resolverConfig.modules`, but keeping in the other options as undocumented or possibly even state they're unsupported options.

Reply:
friendly ping @GertSallaerts. are you still working on this?

Reply:
Updated the docs, what do you think? It's been a busy week on my end too 😅